src/Servers/Kestrel/test/InMemory.FunctionalTests/Http2/Http2TestBase.cs

```diff
@@ -495,7 +496,10 @@ async Task CompletePipeOnTaskCompletion()
         _connectionTask = CompletePipeOnTaskCompletion();
     }
 
-    protected async Task InitializeConnectionAsync(RequestDelegate application, int expectedSettingsCount = 3, bool expectedWindowUpdate = true)
+    protected ConfiguredTaskAwaitable InitializeConnectionAsync(RequestDelegate application, int expectedSettingsCount = 3, bool expectedWindowUpdate = true) =>
+        InnerInitializeConnectionAsync(application, expectedSettingsCount, expectedWindowUpdate).ConfigureAwait(false);
```

Comment:
```suggestion
        InnerInitializeConnectionAsync(application, expectedSettingsCount, expectedWindowUpdate).DefaultTimeout().ConfigureAwait(false);
```

Reply:
I considered doing this when making this change, but I don't like having the timeout at every layer. I almost removed more unnecessary timeouts as part of this change, but I wanted to keep it small.

Adding a default timeout here means that it's more likely to time out at this layer rather than the inner calls to `SendAsync()/FlushAsync()` or `ExpectAsync()/ReadFrameAsync()` which would give you more detail about what actually stalled. Either way, `InitializeConnectionAsync` shows up in the stack trace, but only timing out at the lower level gives a more detailed stack trace.

It's not super important either way because `InitializeConnectionAsync()` rarely times out, but I think it's better to just rely on the existing timeouts in `FlushAsync()` and `ReadFrameAsync()` than litter the in-memory HTTP/2 tests with `.DefaultTimeout()`.

Reply:
Ok. I think only receiving settings inside initializing the connection could hang. Maybe add the timeout there?

That's simple enough to add later if I'm debugging a test locally and it hangs. Doesn't need to block this PR.